fsharp-backend/src/HttpMiddleware/HttpMiddlewareV1.fs

```diff
@@ -0,0 +1,60 @@
+/// Thin, bytes-friendly middleware
+///
+/// This Middleware corresponds with `HTTPBytes` handlers
+[<RequireQualifiedAccess>]
+module HttpMiddleware.HttpMiddlewareV1
+
+open Prelude
+open LibExecution.VendoredTablecloth
+
+module RT = LibExecution.RuntimeTypes
+module Telemetry = LibService.Telemetry
+
+let lowercaseHeaderKeys (headers : HttpHeaders.T) =
+  headers |> List.map (fun (k, v) -> (String.toLowercase k, v))
+
+module Request =
+  let fromRequest
+    (uri : string)
+    (headers : HttpHeaders.T)
+    (query : string)
+    (body : byte array)
+    : RT.Dval =
+    let headers =
+      headers
+      |> lowercaseHeaderKeys
+      |> List.map (fun (k, v) -> RT.DTuple(RT.DStr(k), RT.DStr(v), []))
+      |> RT.DList
+
+    [ "body", RT.DBytes body
+      "queryParams", RT.DStr query
```

Comment:
Should we remove queryParams, like we did in HTTPClientBytes?

Reply:
done